src/locale/localed-util.c

```diff
@@ -148,9 +148,11 @@ int x11_read_data(Context *c, sd_bus_message *m) {
 
         /* Check if there is X11 keyboard data inside vconsole */
         r = vconsole_read_data(c, m);
-        if(r <= 0 && c->x11_layout != NULL && errno != ENOENT) {
-            if(r == 0) log_debug("Found X11 keyboard data inside vconsole, skipping /etc/X11/xorg.conf.d/00-keyboard.conf");
-                return r;
+        if (r < 0 && r != -ENOENT)
```

Comment:
The function `vconsole_read_data()` should not return `-ENOENT`.

Reply:
I see. In that case, how can we check if the `/etc/vconsole.conf` file did not exist? It seems to me this code will skip reading the Xorg keyboard config file if vconsole does not exist

Reply:
@yuwata  I think you're saying `vconsole_read_data` won't return `-ENOENT` because it stats the file before trying to parse it and if the file isn't there, it returns success.

But that stat call is racy of course, so @Diegovsky is right, there is an edge case where `vconsole_read_data` can return `-ENOENT`. It's clear the intention of the code is for `vconsole_read_data` never to return `-ENOENT`, though, so maybe this pull request should ignore that problem ...

I guess we could also take the opportunity to just fix the race (e.g. call `xfopenat(AT_FDCWD, "/etc/vconsole.conf", ..., &f) instead of `stat`, and pass `f` to `parse_env_file` instead of the filename)

Reply:
I like your solution, @halfline but it seems to require some substantial changes to systemd's codebase. Though, I can make a  function that uses a `FILE*` instead of a path and contains all the logic from `parse_env_file` and call it from `parse_env_file`. What do you think @yuwata ?

Reply:
@Diegovsky is this still on your radar ? I don't want to do a plymouth release until this pull request has landed, since I don't want to rely on undocumented features of the config file.

Reply:
Yeah, I'm very busy with college finals, sorry for taking so long to update :(